vscode/src/commands/cache.ts

```diff
@@ -0,0 +1,60 @@
+/*
+  Copyright (c) 2023-2024, Oracle and/or its affiliates.
+
+  Licensed under the Apache License, Version 2.0 (the "License");
+  you may not use this file except in compliance with the License.
+  You may obtain a copy of the License at
+
+     https://www.apache.org/licenses/LICENSE-2.0
+
+  Unless required by applicable law or agreed to in writing, software
+  distributed under the License is distributed on an "AS IS" BASIS,
+  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  See the License for the specific language governing permissions and
+  limitations under the License.
+*/
+import { commands, window } from "vscode";
+import { globalVars } from "../extension";
+import { builtInCommands, extCommands } from "./commands";
+import { ICommand } from "./types";
+import { l10n } from "../localiser";
+import * as fs from 'fs';
+import * as path from 'path';
+
+const deleteCache = async () => {
+    // TODO: Change workspace path to userdir path
+    const storagePath = globalVars.extensionInfo.getWorkspaceStorage()?.fsPath;
+    if (!storagePath) {
+        window.showErrorMessage(l10n.value("jdk.extension.cache.error_msg.cannotFindWrkSpacePath"));
+        return;
+    }
+
+    const userDir = path.join(storagePath, "userdir");
+    if (userDir && fs.existsSync(userDir)) {
+        const yes = l10n.value("jdk.extension.cache.label.confirmation.yes")
+        const cancel = l10n.value("jdk.extension.cache.label.confirmation.cancel")
+        const confirmation = await window.showInformationMessage(l10n.value("jdk.extension.cache.message.confirmToDeleteCache"),
+            yes, cancel);
+        if (confirmation === yes) {
+            const reloadWindowActionLabel = l10n.value("jdk.extension.cache.label.reloadWindow");
+            try {
+                await globalVars.clientPromise.stopClient();
+                globalVars.deactivated = true;
```

Comment:
can we hide this state exposure to owner , extension  , say use a interface to expose required functions like deactivate. global states are code smell in a way , lets try to encapsulate state as much as possible

Reply:
globalVars is a namespace and the variables inside it are required in almost all the files. Even, if we create a class for storing the state, we would need to expose getters and setters for all the variables so essentially it would be the same thing as there is no package protection in typescript. So, I think there is not much difference as it cannot be encapsulated.

Reply:
we can take it up later probably , i will also check. approving for the initial cut